### PR TITLE
Update eslint-plugin dependencies for eslint@8 support

### PR DESCRIPTION
**Pull request by manwaring**
## Description

Update `eslint-plugin` dependencies so that the plugin works with `eslint@8`
- [x] `eslint-plugin-import`
- [ ] `@typescript-eslint` [waiting on v5 changes](https://github.com/typescript-eslint/typescript-eslint/issues/3738)

## Type of change

- [x] `@shopify/eslint-plugin` Patch: Bug (non-breaking change which fixes an issue)
- [ ]  Minor: New feature (non-breaking change which adds functionality)
- [ ]  Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have added a changelog entry, prefixed by the type of change noted above (Documentation fix and Test update does not need a changelog as we do not publish a new version for these changes)

### CODE REVIEW

**Comment by khendry** [comment]
I'm not sure if we have a process or policy against this, but the `@typescript-eslint` issues says that they have an RC available under the tag `rc-v5`. Is it acceptable to close this issue out by using that and then creating another to follow up and integrate the official release once its ready? Or is the a _dependabot_ that would do that for this project?

File: dev.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: web-configs
 up:
   - node:
-      version: v12.14.0
+      version: v12.22.0
```
Comment (manwaring) [inline dev.yml:4]: Updated ESLint required v12.22.0 or higher so bumped

**Comment by BPScott** [comment]
I'll be taking on updating eslint-plugin so that it uses the newest versions of the various plugins, and supports eslint v8 in the next few days. I'll be doing the same thing with the stylelint-plugin to make sure it has support for stylelint v14 too.

**Comment by manwaring** [comment]
Thanks @BPScott - sorry for the delay on my end, I was out unexpectedly for bereavement leave and have been catching up on lots of things this week.  If it's helpful please continue using the work in this PR (but no worries if you prefer to start fresh)

**Review by khendry — APPROVED** [review]
lgtm

**Comment by BPScott** [comment]
Closing in favour of #305

**Pull request closed without merging**